stake-pool/program/src/processor.rs

```diff
@@ -1315,9 +1315,17 @@ impl Processor {
                 vote_account_address.as_ref(),
                 ValidatorStakeInfo::memcmp_pubkey,
             );
-            if maybe_validator_stake_info.is_none() {
-                msg!("Validator for {} not present in the stake pool, cannot set as preferred deposit account");
-                return Err(StakePoolError::ValidatorNotFound.into());
+            match maybe_validator_stake_info {
+                Some(vsi) => {
+                    if vsi.status == StakeStatus::ReadyForRemoval {
```

Comment:
I think we should also avoid a validator stake in the `DeactivatingTransient` state, so how about this becomes:
```suggestion
                    if vsi.status != StakeStatus::Active {
```